streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/TopologyTest.scala

```diff
@@ -457,9 +457,10 @@ class TopologyTest {
     assertEquals(getTopologyScala.build(props).describe.toString, getTopologyJava.build(props).describe.toString)
   }
 
-  private class SimpleProcessor private[TopologyTest] (val valueList: util.List[String])
-      extends AbstractProcessor[String, String] {
-    override def process(key: String, value: String): Unit =
-      valueList.add(value)
+  private class SimpleProcessorSupplier private[TopologyTest] (val valueList: util.List[String])
+      extends ProcessorSupplier[String, String, Nothing, Nothing] {
+
+    override def get(): Processor[String, String, Nothing, Nothing] =
+      (record: api.Record[String, String]) => valueList.add(record.value())
```

Comment:
I went ahead and changed the test to demonstrate that SAM conversion doesn't work with these methods, so we have to declare a supplier instead of a `() => processor`.